optaweb-employee-rostering-frontend/src/ui/components/WeekPicker.tsx

```diff
@@ -40,28 +35,71 @@ function getLastDayInWeek(dateInWeek: Date): Date {
   return moment(dateInWeek).endOf('week').toDate();
 }
 
-export default class WeekPicker extends React.Component<WeekPickerProps, WeekPickerState> {
-  constructor(props: WeekPickerProps) {
-    super(props);
-    this.state = { isOpen: false };
-  }
-
+export default class WeekPicker extends React.Component<WeekPickerProps> {
   goToCurrentWeek() {
     this.goToWeekContaining(new Date());
   }
 
   goToWeekContaining(date: Date) {
     if (this.props.minDate && getFirstDayInWeek(date) < getFirstDayInWeek(this.props.minDate)) {
       this.goToWeekContaining(this.props.minDate);
-    } else if (this.props.maxDate && getLastDayInWeek(date) > getLastDayInWeek(this.props.maxDate)) {
+      return;
+    } if (this.props.maxDate && getLastDayInWeek(date) > getLastDayInWeek(this.props.maxDate)) {
       this.goToWeekContaining(this.props.maxDate);
+      return;
     }
     this.props.onChange(getFirstDayInWeek(date), getLastDayInWeek(date));
-    this.setState({ isOpen: false });
   }
 
   render() {
-    const locale = moment.locale();
+    type DatePickerCustomInputProps = { value?: Date; onClick?: () => void };
+
+    // (the stateless functional component is defined in a class component, where
+    //  this = the class this, so it okay to use "this" here)
+    /* eslint-disable react/no-this-in-sfc */
+    const DateRangeInputElement = React.forwardRef<HTMLDivElement, DatePickerCustomInputProps>(
+      (dateRangeInputProps, ref) => (
+        <div
+          ref={ref}
+          className="pf-c-form-control"
+          style={{
+            display: 'grid',
+            height: 'auto',
+            gridTemplateColumns: '1fr auto auto',
+            alignItems: 'center',
+          }}
+        >
+          <Text
+            onClick={dateRangeInputProps ? dateRangeInputProps.onClick : undefined}
```

Comment:
Do you know how the `onClick` works? I wonder if it is possible to make it work like a toggle. First click should open the date picker, second click should close it. If I click the text accidentally or change my mind and decide not to select a date it is now impossible to hide the data picker with another click at the same place. For me it's a bad user experience.

Reply:
I managed to enable toggle behaviour by getting the DatePicker ref and calling `setOpen`, keeping track of the `isOpen` state in the custom input component.

Reply:
Great, thanks!